napalm/eos/eos.py

```diff
@@ -138,9 +138,11 @@ def is_alive(self):
     def _lock(self):
         if self.config_session is None:
             self.config_session = 'napalm_{}'.format(datetime.now().microsecond)
+        else:
+            raise SessionLockedException('Session is already in use: {}'.format(self.config_session))
         sess = self.device.run_commands(['show configuration sessions'])[0]['sessions']
         if [k for k, v in sess.items() if v['state'] == 'pending' and k != self.config_session]:
-            raise SessionLockedException('Session is already in use')
+            raise SessionLockedException('Session is already in use: {}'.format(k))
```

Comment:
The variable k won't be known outside of the list comprehension. So that will need to be corrected.